watertap/unit_models/cstr_injection.py

```diff
@@ -0,0 +1,338 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Modified CSTR model which includes terms for injection of species/reactants.
+
+This is copied the standard IDAES CSTR with the addition of mass trasnfer terms.
```

Comment:
Without having gone through the CSTR class to sort out the details myself, could this have just been inherited from CSTR, followed by setting has_mass_transfer True for the control volume?

Reply:
Not without changing the CSTR model, and I decided that was not appropriate in this case. However, I think we should generalize this model a bit more to also cover the anaerobic digestor case (and any liquid-phase reactor with vapour headspace we might have).